client/client.go

```diff
@@ -0,0 +1,150 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-delegated-routing/parser"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/multiformats/go-multiaddr"
+
+	//"github.com/ipld/go-ipld-prime/codec/dagjson"
+	"io"
+	"net/http"
+)
+
+type Option func(*client) error
+
+type client struct {
+	client *http.Client
+	endPoint string
+}
+
+func WithHTTPClient(hc *http.Client) Option {
+	return func(c *client) error {
+		c.client = hc
+		return nil
+	}
+}
+
+func New(endpoint string, opts...Option) (*client, error) {
+	c := &client{endPoint: endpoint, client: http.DefaultClient}
+	for _, o := range opts {
+		if err := o(c); err != nil {
+			return nil, err
+		}
+	}
+	return c, nil
+}
+
+func (c *client) FindProviders(ctx context.Context, cid cid.Cid) ([]peer.AddrInfo, error) {
```

Comment:
This interface is not async. Did you intend that?

Reply:
This was mostly for prototyping. It's both easy and it's what the hydras will need since they need the data synchronously before sending back a response.

Making an async function here makes sense too though as we can always turn async into sync